docs/projects/tad/adrs/0005-ai-verify-technical-tests.md

```diff
@@ -0,0 +1,40 @@
+# TAD-0005 Use AI Verify for technical tests
+
+## Context
+
+The AI Verify project is set up in a modular way, and the technical tests is one of the modules. The AI Verify
+team is developing a feature which makes it possible to run the technical tests using an API, by using
+a python file calling a method to run a test and providing the required configuration, like which
+model and dataset to use and some test specific configuration. The result of the test are returned in a JSON format,
+which can be processed as we want, like write it to a file or store it in a database.
+
+### Pros
+
+* We have several technical tests we can use of the shelf.
+* Tests are set up in a generic way using interfaces which allows others, like ourselves, to create their own plugins.
+* Loading models, pipelines and data is done through the AI Verify toolkit, which does validation
+ and support checks.
+
+### Cons
+
+* The result of a test is dependent on how AI Verify has implemented it.
+* Adding our own plugins may require adding AI Verify front end blocks we don't use ourselves.
+* We are dependent on the AI Verify ecosystem for supported models and data formats.
+
+## Assumptions
+
+* We can wrap the API and other AI Verify requirements in a docker image.
+* We can run the docker image independently where we only have to provide the model, datasets and other
+ required configuration to run a test.
+
+## Decision
+
+We will use the technical tests of AI Verify for the TAD tool.
+
+## Risks
+
+We are tied to the AI Verify ecosystem.
```

Comment:
We are not fully tied to the AI verify ecosystem, just for specific technical tests. Probably if tests are missing we should maybe write our own wrapper for them if we don't want to immediately make a AIVerify Plugin

Reply:
True. What this intended to state, as ecosystem, is that besides the tests, we also require the test-engine-core etc. modules. So, to load a model or data, we depend on if / how AI-Verify has implemented this. I will make that more clear.